### PR TITLE
[IOPLT-556] Add process-activations-changes

### PR DESCRIPTION
**Pull request by datalek**
#### List of Changes

- Add process-activation-request use-case
- Rename activation-job file to process-activation-job
- Rename activation-job file to process-activation-changes
- Add `SubscriptionHistoryReader` with `getLatest` method

#### Motivation and Context

Updates made to the activations container should be reflected in the subscription-history container.

#### How Has This Been Tested?

Unit tests

#### Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] I have committed only changes relevant to the task.
- [x] I have minimized the number of changes.
- [ ] My changes are about only one task or issue.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/functions-subscription/src/use-cases/__tests__/process-activation-request.test.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    const actual =
+      await processActivationRequest(anActivationRequest)(testEnv)();
+    const expected = E.left(new Error('Subscription History not found'));
+
+    expect(actual).toStrictEqual(expected);
+    expect(mockEnv.subscriptionHistoryReader.getLatest).toBeCalledTimes(1);
+    expect(mockEnv.subscriptionHistoryWriter.insert).toBeCalledTimes(0);
+  });
+
+  it('should fail if update subscription raise an error', async () => {
```
Comment (kin0992) [inline apps/functions-subscription/src/use-cases/__tests__/process-activation-request.test.ts:60]:
```suggestion
  it('should fail if insert subscription raise an error', async () => {
```

File: apps/functions-subscription/src/use-cases/__tests__/process-activation-request.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      TE.left(error),
+    );
+
+    const actual =
+      await processActivationRequest(anActivationRequest)(testEnv)();
+
+    expect(actual).toStrictEqual(E.left(error));
+  });
+
+  it('should fail if get subscription raise an error', async () => {
```
Comment (kin0992) [inline apps/functions-subscription/src/use-cases/__tests__/process-activation-request.test.ts:81]:
```suggestion
  it('should fail if get latest subscription history raise an error', async () => {
```

File: apps/functions-subscription/src/domain/subscription-history.ts [hunk 3]
```diff
@@ -57,6 +64,18 @@ export const makeSubscriptionHistory = (subscription: Subscription) => {
   );
 };
 
+export const updateSubscriptionHistory = (
```
Comment (kin0992) [inline apps/functions-subscription/src/domain/subscription-history.ts:67]: **note**: we are not updating the subscription history, but we are creating a new version. maybe we could rename this function accordingly.
what do you think?

Comment (datalek) [reply apps/functions-subscription/src/domain/subscription-history.ts:67]: Yes, what about renaming it to `makeSubscriptionHistoryNextVersion`?

Comment (kin0992) [reply apps/functions-subscription/src/domain/subscription-history.ts:67]: Sounds good to me!

Comment (datalek) [reply apps/functions-subscription/src/domain/subscription-history.ts]: Done ✅

**Review by kin0992 — APPROVED** [review]

**Pull request merged**